src/main/java/com/google/devtools/build/lib/analysis/LocationExpander.java

```diff
@@ -387,39 +397,53 @@ static ImmutableMap<String, LocationFunction> allLocationFunctions(
                 locationMap,
                 execPaths ? PathType.EXEC : PathType.LOCATION,
                 legacyExternalRunfiles,
-                EXACTLY_ONE))
+                EXACTLY_ONE,
+                FULL_PATH))
         .put(
             "locations",
             new LocationFunction(
                 root,
                 locationMap,
                 execPaths ? PathType.EXEC : PathType.LOCATION,
                 legacyExternalRunfiles,
-                ALLOW_MULTIPLE))
+                ALLOW_MULTIPLE,
+                FULL_PATH))
         .put(
             "rootpath",
             new LocationFunction(
-                root, locationMap, PathType.LOCATION, legacyExternalRunfiles, EXACTLY_ONE))
+                root, locationMap, PathType.LOCATION, legacyExternalRunfiles, EXACTLY_ONE, FULL_PATH))
         .put(
             "rootpaths",
             new LocationFunction(
-                root, locationMap, PathType.LOCATION, legacyExternalRunfiles, ALLOW_MULTIPLE))
+                root, locationMap, PathType.LOCATION, legacyExternalRunfiles, ALLOW_MULTIPLE, FULL_PATH))
+        .put(
+            "rootpath_dirname",
+            new LocationFunction(
+                root, locationMap, PathType.LOCATION, legacyExternalRunfiles, EXACTLY_ONE, DIRNAME_ONLY))
         .put(
             "execpath",
             new LocationFunction(
-                root, locationMap, PathType.EXEC, legacyExternalRunfiles, EXACTLY_ONE))
+                root, locationMap, PathType.EXEC, legacyExternalRunfiles, EXACTLY_ONE, FULL_PATH))
         .put(
             "execpaths",
             new LocationFunction(
-                root, locationMap, PathType.EXEC, legacyExternalRunfiles, ALLOW_MULTIPLE))
+                root, locationMap, PathType.EXEC, legacyExternalRunfiles, ALLOW_MULTIPLE, FULL_PATH))
+        .put(
+            "execpath_dirname",
+            new LocationFunction(
+                root, locationMap, PathType.EXEC, legacyExternalRunfiles, EXACTLY_ONE, DIRNAME_ONLY))
         .put(
             "rlocationpath",
             new LocationFunction(
-                root, locationMap, PathType.RLOCATION, legacyExternalRunfiles, EXACTLY_ONE))
+                root, locationMap, PathType.RLOCATION, legacyExternalRunfiles, EXACTLY_ONE, FULL_PATH))
         .put(
             "rlocationpaths",
             new LocationFunction(
-                root, locationMap, PathType.RLOCATION, legacyExternalRunfiles, ALLOW_MULTIPLE))
+                root, locationMap, PathType.RLOCATION, legacyExternalRunfiles, ALLOW_MULTIPLE, FULL_PATH))
+        .put(
+            "rlocationpath_dirname",
```

Comment:
This is problematic as most runfiles libraries won't support this (with the manifest representation of runfiles, it requires them to construct a full path trie, not just a regular map for lookups). `rootpath_dirname` could be used instead on Unix.

Reply:
Thanks! I'll remove `$(rlocationpath_dirname)` from this PR.